benchmarks/profiler/utils/utils.py

```
@@ -0,0 +1,136 @@
+        for line in ps_output.splitlines():
+            if "python" in line.lower():
+                parts = line.split()
+                if len(parts) >= 2:
+                    try:
+                        pid = int(parts[1])
+                        if pid != current_pid:  # Exclude current process
+                            os.kill(pid, signal.SIGKILL)
+                    except ValueError:
+                        continue
+    except Exception as e:
+        logger.error(f"Error killing Python processes: {e}")
```

Pylint (3.3.7)

[refactor] Too many nested blocks (6/5)

(R1702)